Anahit dev

anahitH opened the pull request:
Implemented features and corresponding activities
    User login, SigninActivity
    User signup, SignupActivity
    User logout, from UserActivity option menu
    User personal data retrieving, UserActivity. The content of AttendancesActivity is moved to UserActivity
    Courses retrieving, CoursesActivity, opens from option menu in UserActivity
    Groups retrieving (partial). Shows all groups of a selected course, opens from CoursesActivity.

Minor changes in QRCodeActivity. Modified AsynchTask of qr code retrieval to extent BaseAsynchTask, so it will have loading bar. Moved the part of getting qr image from the token to background thread, so that ui does not hang.

anahitH committed:
Implemented user login.
Added aat datastore package as a library to android.

anahitH committed:
Implemnted groups retrieving in CourseActivity

anahitH committed:
Added parsing group id from string to long

anahitH committed:
Merge branch 'develop' into anahit_dev

anahitH committed:
Added new activities

anahitH committed:
Implemented endpoints connection from some activities

anahitH committed:
Added dependencies for jackson in android gradle

anahitH committed:
Merge branch 'develop' into anahit_dev

anahitH committed:
Fixed version of jackson libraries

anahitH committed:
Merge branch 'develop' into anahit_dev

anahitH committed:
Implemeted user retrieving

anahitH committed:
Renamed packages in Android part, so that do not have many conflict with the develop later
Added Courses retrieveing
Added logout

anahitH committed:
Merge branch 'develop' into anahit_dev

anahitH committed:
Modified client request to use EndpointsUtils.
Made minor changes in ui

djwessel commented on AAT_android/app/src/main/java/com/ase/aat_android/activities/CourseActivity.java:
```diff
@@ -12,29 +12,113 @@
+            ObjectMapper mapper = new ObjectMapper();
+            List<Object> groupObjects = null;
+            try {
+                // Throws exception.
```
If this is all commented out, do we even need it?

djwessel commented on AAT_android/app/src/main/java/com/ase/aat_android/activities/CoursesActivity.java:
```diff
@@ -0,0 +1,220 @@
+        Thing object = new Thing.Builder()
+                .setName("Courses Page") // TODO: Define a title for the content shown.
+                // TODO: Make sure this auto-generated URL is correct.
+                .setUrl(Uri.parse("http://[ENTER-YOUR-URL-HERE]"))
```
What is the purpose of this auto generated code? If we are using it, should we change the url?

anahitH committed:
Fixed issue with getting group lists.
Implemented group registration and de-registration.
Added wrapper classes for datastore Group and Course entyties.

anahitH committed:
Merge branch 'develop' into anahit_dev

anahitH committed:
Added course_id to attendance deleting endpoint

yemramirezca commented on AAT_android/app/src/main/java/com/ase/aat_android/activities/CoursesActivity.java:
```diff
@@ -0,0 +1,169 @@
+            ObjectMapper mapper = new ObjectMapper();
+            List<Object> courseObjects = null;
+            try {
+                courseObjects = mapper.readValue(coursesRetRes.get().getText(), new TypeReference<List<com.aat.datastore.Course>>() {
```
I tried before something like: 
List<Group> courseObjects =  mapper.readValue(coursesRetRes.get().getText(), new TypeReference<List<Object>>() 

In that case you wouldn't need the method ArrayList<Course> retrieveCourses(List<Object> courseObjects).

yemramirezca commented on AAT_android/app/src/main/java/com/ase/aat_android/data/Course.java:
```diff
@@ -0,0 +1,47 @@
+ */
+
+public class Course implements Serializable{
+    com.aat.datastore.Course course;
```
Why is necessary a new Course class which only contains a Course?

anahitH replied:
This class is used when passing Course object from CoursesActivity to CourseActivity.
Before all fields of datastore Course object were put into the extras (as datastore Course is not implementing Serializable, it is not possible to put its object into an extra).
This class provides easier way to Course object.

anahitH committed:
Addressed reviews.
Made some minor modifications
